app/serializers/api/v2/csv/csv_serializer.rb

```diff
@@ -0,0 +1,56 @@
+module Api
+  module V2
+    module Csv
+      module CsvSerializer
+        extend ActiveSupport::Concern
+
+        def initialize(serializables, options = {})
+          @serializables = serializables
+          @options = options
+        end
+
+        def serializable_array
+          @serializables.each_with_object(initial_rows) do |serializable, acc|
+            data_row = columns.map do |column|
+              serializable.public_send(column)
+            end
+
+            acc << data_row
+          end
+        end
+
+        def serialized_csv
+          serialized = CSV.generate do |csv|
+            serializable_array.each do |row|
+              csv << row
+            end
+          end
+
+          serialized.html_safe
```

Comment:
A CSV shouldn't need to be marked as `.html_safe` since its not related to HTML? This would only be necessary if its being processed by ERB ?

I think `render content: ....` will treat the supplied string as an inline erb template, which may be why this was required - if so `render plain: ` should avoid the use of ERB but using `send_data` would be simpler still.

Reply:
Ah, I suspect that's a hangover, yeah